testing/python/collect.py

```diff
@@ -1583,3 +1583,43 @@ def test_package_collection_infinite_recursion(testdir):
     testdir.copy_example("collect/package_infinite_recursion")
     result = testdir.runpytest()
     result.stdout.fnmatch_lines("*1 passed*")
+
+
+def test_package_with_modules(testdir):
```

Comment:
@jonozzz 
This test unfortunately does not fail for me before the fix, i.e. it does not test the HACK really (which I was looking into to remove/fix properly).
I've copied the test file and then run it against 4d8903f (3.7.2~10).

Reply:
Oh, sorry - confused it with testing/test_collection.py.